### DISCUSSION TITLE
@vercel/postgres appears to break in production

### DISCUSSION THREAD

**Discussion by 619** [vercel/community#3210] [Help] [answered]
### Problem Description

I started a project consisting of a single API function that imports @vercel/postgres (import { sql } from '@vercel/postgres';
) as it's only import where the only package I ever installed with npm was @vercel/postgres (npm i @vercel/postgres). It works on localhost when I run 'vercel dev', but when it is pushed to production on vercel, the API function fails with this error:

```
2023-07-11T14:55:44.204Z	undefined	ERROR	Cannot find module '/var/task/node_modules/@vercel/postgres/dist/index-node.cjs'
2023-07-11T14:55:44.204Z	undefined	ERROR	Did you forget to add it to "dependencies" in `package.json`?
RequestId: a4dc0c40-e5ff-4a94-befb-dbdc8198d56d Error: Runtime exited with error: exit status 1
Runtime.ExitError
```

I am using ^0.4.0 version of @vercel/postgres. What could be the cause of such an error?

**Comment by amyegan** [comment]
Hi @619. Can you share the `package.json` or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? I'm not able to replicate the error with my test project.

**Reply by Bejan512** [reply]
I'm currently having the same problem.

This is the error:
```
2023-07-17T21:42:41.396Z	undefined	ERROR	Cannot find module '/var/task/node_modules/@vercel/postgres/dist/index-node.cjs'
2023-07-17T21:42:41.396Z	undefined	ERROR	Did you forget to add it to "dependencies" in `package.json`?
RequestId: e417346c-cfbb-40cd-a32a-867ae9fcc7e0 Error: Runtime exited with error: exit status 1
Runtime.ExitError

```

package.json:

```
{
  "name": "simple-landing-page",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@vercel/postgres": "^0.4.0"
  }
}
```

api/submit-email.js:

```
import { db } from '@vercel/postgres';

export default async function handler(request, response) {
  const client = await db.connect();
  const email = request.body.email;
  await client.sql`INSERT INTO emails (email) VALUES (${email})`;
  response.status(200).json({
    message: "Email submitted successfully!"
  });
}
```
Thanks for any help. If I manage to find the solution I will update this post.

**Reply by Bejan512** [reply, marked as answer]
It seems i've put the cart before the horse with asking this question.

Looks like it was as simple as my package.json file missing the line - "type": "module" - to allow ES6 import syntax.
